src/ScottPlot4/ScottPlot/Plottable/Image.cs

```diff
@@ -23,6 +23,16 @@ public class Image : IPlottable
         /// </summary>
         public double Y { get; set; }
 
+        /// <summary>
+        /// Width of the image, (0 means no scaling)
+        /// </summary>
+        public double Width { get; set; } = 0;
```

Comment:
I don't know if I like this, for a few reasons:
- 0 is a weird sentinel value, to me width of zero means width of zero, not the width of the underlying bitmap
    - Could change it to `double?` and null would be your sentinel value. A width of null removes ambiguity, it's not a number so it cannot be a normal width.
- As a user, I expect setting `img.Width` to change the image width, but instead, the image starts growing/shrinking as I zoom in or out
    - This could be fixed by a rename, for example `img.WidthInAxisUnits`
- As a user, I (might) expect reading `img.Width` to return the size in pixels
    - This could also be fixed by a rename

I don't particularly like the name `WidthInAxisUnits`, so that part may still need some work. It's still not intuitive that nulling out that property would cause the width to fallback to the width of the underlying bitmap, so maybe there's some merit to something like this:

```cs
public enum LengthUnit
{
    Pixel,
    Axis
}
```

Then you could keep the name as `Width`, and depending on the selected unit it would mean either width in pixels or in the axis space. This seems clunky to me, and I actually think it's less clear if a user doesn't see the `LengthUnit` property. Maybe there's an option in the middle?

Reply:
There is also an option for the user to have the width be axis dependent and the height be independent. I assume there's no use-case for this, so my question is should we prevent it?

If we don't, there is an opportunity for us to introduce bugs where we check only that the width is axis dependent, and then assume that the height is as well. For example:

```cs
if (Width is null) {
    width = CalculateWidth();
    height = CalculateHeight();
}
```

When in reality it should be:

```cs
if (Width is null) {
    width = CalculateWidth();
}

if (Height is null) {
    height = CalculateHeight();
}
```

We could prevent this by checking for it and throwing, but there's actually a rather elegant alternative that doesn't rely on run-time checks. If we put the width and height into a struct we can require them be either both null, or both non-null:

```cs
public struct Dimensions
{
    public Width { get; set; }
    public Height { get; set; }
}

public class Image : IPlottable
{
   .
   .
   .
   public Dimensions? DimensionsInAxisSpace { get; set; }
}
```

Here the user is not even capable of nulling width without nulling out the whole struct. This prevents one axis from being axis dependent while the other is not.

Reply:
I finally chosen separate width and height. Because both nullable struct and class requires the user to create new object first, this is not stright forward.